### PR TITLE
Improve UX for reuse of certificates [full ci]

### LINKED ISSUE #2197: Add user authentication to the personality
The default docker mechanism is TLSVERIFY. This would entail:
a. modification to vic-machine to supply CA certificate to validate clients

There is also the extended docker authorization plugin mechanism which presumably requires that the client send user credentials (or session identifier) with each request. I do not know the details of this mechanism and a quick search of the remote API 1.24 docs didn't make it clear.

Tagged as vic-machine because any approach will require vic-machine changes for configuration. Estimate is for TLSVERIFY - see vicadmin for current mechanism.

https://bugzilla.eng.vmware.com/show_bug.cgi?id=1727625

### PR DESCRIPTION
**Pull request by hickeng**
Adds a --cert-path option, used both for sourcing existing certs and
saving generated certs.

If existing certs are found, those will be used - this avoids the scenario
of redeploying a VCH and overwriting the generated (and potentially
distributed) certificates.
Various checks are performed to ensure certificate validity and attempts
to provide better diagnostics with cert errors.

Now uses the system root pool to validate server cert as well as the
generated CA when checking API liveliness.

Fixes #2197

### CODE REVIEW

File: cmd/vic-machine/create/create.go [hunk 1]
```diff
@@ -886,46 +1013,51 @@ func (c *Create) generateCertificates(ca bool) ([]byte, *certificate.KeyPair, er
...
-	if err = ckp.SaveCertificate(); err != nil {
-		log.Errorf("Failed to save client certificates: %s", err)
-		return nil, nil, err
-	}
+	if client {
+		log.Infof("Generating client certificate/key pair - private key in %s", c.ckey)
+		ckp := certificate.NewKeyPair(c.ccert, c.ckey, nil, nil)
+		err = ckp.CreateClientCertificate(c.cname, c.org, c.keySize, cakp)
+		if err != nil {
+			log.Errorf("Failed to generate server certificates: %s", err)
```
Comment (andrewtchin) [inline cmd/vic-machine/create/create.go]: should this be client certificates?

**Comment by andrewtchin** [comment]
lgtm

File: cmd/vic-machine/create/create.go [hunk 2]
```diff
@@ -777,81 +815,170 @@ func (c *Create) processProxies() error {
...
 	}
 
 	var keypair *certificate.KeyPair
-	if c.cert != "" && c.key != "" {
-		log.Infof("Loading certificate/key pair - private key in %s", c.key)
-		keypair = certificate.NewKeyPair(c.cert, c.key, nil, nil)
-
-		if err := keypair.LoadCertificate(); err != nil {
+	// default names
+	skey := fmt.Sprintf("./%s/%s", c.certPath, serverKey)
```
Comment (fdawg4l) [inline cmd/vic-machine/create/create.go]: Will this bale on $WINDOWS?  Might want to use `path.Join()` if it will.

**Comment by fdawg4l** [comment]
LGTM

**Comment by hickeng** [comment]
poke

**Comment by hickeng** [comment]
poke

**Pull request merged**